src/main/kotlin/com/avail/descriptor/pojos/RawPojoDescriptor.kt

```diff
@@ -0,0 +1,272 @@
+ * Avail programs will use [typed&#32;pojos][PojoDescriptor] universally, but
+ * the implementation mechanisms frequently require raw pojos (especially for
+ * defining [pojo&#32;types][PojoTypeDescriptor]).
+ *
+ * @constructor
+ *
+ * @param mutability
+ *   The [mutability][Mutability] of the new descriptor.
+ * @param javaObject
+ *   The actual Java [Object] represented by the sole [AvailObject] that will
+ *   use the new descriptor.
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ * @see EqualityRawPojoDescriptor
+ */
+open class RawPojoDescriptor protected constructor(
+	mutability: Mutability,
+	val javaObject: Any?
+) : Descriptor(mutability, TypeTag.POJO_TAG, null, null
+) {
+	@AvailMethod
+	override fun o_Equals(self: AvailObject, another: A_BasicObject): Boolean =
+		another.equalsRawPojoFor(self, javaObject)
+
+	@AvailMethod
+	override fun o_EqualsEqualityRawPojo(
+		self: AvailObject,
+		otherEqualityRawPojo: AvailObject,
+		otherJavaObject: Any?
+	) = false
+
+	@AvailMethod
+	override fun o_EqualsRawPojoFor(
+		self: AvailObject,
+		otherRawPojo: AvailObject,
+		otherJavaObject: Any?
+	): Boolean {
+		when {
+			javaObject !== otherJavaObject -> return false
+			self.sameAddressAs(otherRawPojo) -> return true
+			!isShared -> self.becomeIndirectionTo(otherRawPojo)
+			!otherRawPojo.descriptor().isShared ->
+				otherRawPojo.becomeIndirectionTo(self)
+		}
+		return true
+	}
+
+	@AvailMethod
+	override fun o_Hash(self: AvailObject): Int {
+		// This ensures that mutations of the wrapped pojo do not corrupt hashed
+		// Avail data structures.
+		return System.identityHashCode(javaObject) xor 0x277AB9C3
+	}
+
+	@AvailMethod
+	override fun o_IsRawPojo(self: AvailObject): Boolean = true
+
+	@AvailMethod
+	override fun <T> o_JavaObject(self: AvailObject): T? =
+		nullableCast<Any?, T?>(javaObject)
+
+	@AvailMethod
+	override fun o_Kind(self: AvailObject): A_Type = Types.RAW_POJO.o()
+
+	/**
+	 * Replace the descriptor with a newly synthesized one that has the same
+	 * [javaObject] but is [immutable][Mutability.IMMUTABLE].
+	 */
+	@AvailMethod
+	override fun o_MakeImmutable(self: AvailObject): AvailObject {
+		if (isMutable) {
+			self.setDescriptor(
+				RawPojoDescriptor(Mutability.IMMUTABLE, javaObject))
+		}
+		return self
+	}
+
+	/**
+	 * Replace the descriptor with a newly synthesized one that has the same
+	 * [javaObject] but is [shared][Mutability.SHARED].
+	 */
+	@AvailMethod
+	override fun o_MakeShared(self: AvailObject): AvailObject {
+		if (!isShared) {
+			self.setDescriptor(
+				RawPojoDescriptor(Mutability.SHARED, javaObject))
+		}
+		return self
+	}
+
+	override fun o_MarshalToJava(
+		self: AvailObject,
+		ignoredClassHint: Class<*>?
+	): Any? = javaObject
+
+	override fun o_SerializerOperation(self: AvailObject): SerializerOperation =
+		when (javaObject) {
+			null -> SerializerOperation.RAW_POJO_NULL
+			is Class<*> -> {
+				val o = cast<Any, Class<*>>(javaObject)
+				when {
```

Comment:
open brace on new line

Reply:
Folded "val o..." and "o.isPrimitive" into when's argument, so I removed the braces.